### ISSUE TITLE
Support NA-filling

### ISSUE THREAD

**Issue by Ironholds** [ropensci/gender#49]
At the moment if I pass 50 names in, and 10 don't match, I'll get 40 rows back. This can be a problem for explicitly getting a row per entry. I'd propose adding a `fill_missing` parameter, set to `FALSE` by default, that allows one to explicitly include rows for names with missing data. If that's okay I'll write the patch myself!

**Comment by lmullen** [comment]
Hmm. My main assumption is that people are taking the resulting data frame and joining it back to the original data (as in the vignette: https://cran.rstudio.com/web/packages/gender/vignettes/predicting-gender.html) Do you think there are different use cases where one needs the explicit missing values? If so, feel free to write the patch.

In general, the interface of this package and probably the implementation needs to be rethought from the group up, but that's fairly low on my list of things to do.

**Comment by Ironholds** [comment]
Yeah, I'm definitely here for wider API changes to if you want me to noodle on a structure and come up with a proposal?

**Comment by lmullen** [comment]
If you're up for it, that would be great. 

In general I think there are several kinds of data that might go in. Let's assume that it should work with either data frames or character vectors.

The simplest input is that there is a list of names and a year or range of years associated with all of them.

The more complex input is that there is a list of names and a year or range of years associated with each one individually.

The most complex input is that there is a list of names and a year or range of years associated with each one individually, and that the range of years are incompatible with the different methods.

I think it might be desirable for the function to pick the method (ssa, ipums) from the input year unless explicitly told otherwise.

Thanks very much for taking this on.
